jssource/jsmin.php

```diff
@@ -45,4 +45,50 @@ protected function jsParser() {
         require_once('jssource/Minifier.php');
         return Minifier::minify($this->text);
 	}
+
+    /**
+     * Join and minify JS files
+     *
+     * @param array $jsFiles an 'array' of js files
+     *
+     * @author Jose C. Massón <jose@gcoop.coop>
+     *
+     * @return Minified JS file path
+     */
+    static public function joinAndMinifyJSFiles($jsFiles)
+    {
+        $target = SugarThemeRegistry::current()->getJSPath()
+                . '/' .
+                sha1(implode('|', $jsFiles)) . '.js';
+
+        if (!is_file(sugar_cached($target))) {
+            $customJSContents = '';
+
+            foreach ($jsFiles as $jsFileName) {
+                $jsFileContents = '';
+
+                if (is_file($jsFileName)) {
+                    $jsFileContents .= sugar_file_get_contents($jsFileName);
+                }
+
+                if (empty($jsFileContents)) {
+                    LoggerManager::getLogger()->warn(
+                        translate('ERR_FILE_EMPTY') .': '. "{$jsFileName}"
+                    );
+                }
+                $customJSContents .= $jsFileContents;
+            }
+
+            $customJSPath = create_cache_directory($target);
+
+            if ((!inDeveloperMode()) && (!is_file($customJSPath))) {
```

Comment:
Too many extra parentheses. The following expressions means exactly same thing but simple:
```
-> !inDeveloperMode() && !is_file($customJSPath)
-> !(inDeveloperMode() || is_file($customJSPath))
```
(also I don't really understand why developer mode check is necessary here?)

Reply:
Ok.
We check If you are in developer mode because when developing JS you don't want to have JS concatenated and minifyed...It is not easy to debug JS.